File: Recursion/javascript/factorial.js
function factorial(num) {
  if(num < 0) {
    console.log("number should be positive")
  }

  if(num === 1) {
    return 1
  }

  if(num > 1) {
    return num * factorial(num - 1)
  }
}

//////////////////////////////

function factorial2(n) {
  let result = 1
  while(n > 0) {
    result *= n
    n--
  }
  return result
}

//////////////////////////////

function factorial3(n) {
  let result = 1;
  for(let i=1 ; i<=n ; i++) {
    result = result * i
  }
  return result
}

console.log(factorial(4))
console.log(factorial2(4))
console.log(factorial3(4))

File: Recursion/factorial.py
def factorial(n):
    result = 1
    while n > 0:
        result = result * n
        n -= 1

    return result

print(factorial(4))

#######################

def factorial2(n):
    result = 1
    for i in range(1, n+1):
        result = result * i
    
    return result

print(factorial2(4))

########################

def factroial3(n):
    if n == 1:
        return n
    if(n > 1):
        return n * factroial3(n-1)

print(factroial3(4))